expression.go

```diff
@@ -317,3 +317,19 @@ func (expr GreaterEqualsExpr) RootRefs() []FieldExpr {
 	out = rootSetAdd(out, expr.Rhs.RootRefs()...)
 	return out
 }
+
+type MatchesExpr struct {
```

Comment:
Any reason for using "Matches" as the keyword rather than "Like" which would be more idiomatic to existing terminology used in Couchbase (specifically N1QL and SQL++).

Reply:
One of the concerns brought up with LIKE is that it's possible users will want to use N1QL LIKE syntax, which isn't regex. So I thought MATCHES would be more universally linked with regex.
From a product perspective, it does make sense to use LIKE since it's already present, but users may gawk when the parser barfs back a syntax error or internally treats it as a string literal.

Reply:
I'll change it to LIKE, per @brett19 suggestion unless otherwise noted.

Reply:
I think the reasoning is relatively reasonable (I forgot that N1QL LIKE is not regexp), but MATCHES is a bit generic, is there some other term that might be more clear without overloading the term?

Reply:
I think what we can do is keep it as a LikeExpr... and for now, the RHS leaf node can be a RegexExpr node. So we know that the operation is a regex.
Should we support N1QL matching the future, the RHS leaf node can be something else. The Like operation here can be overloaded.
Is there a reason why you don't like overloading?

Reply:
I think having a RegexExpr node on the RHS should make sense.  Overloading is just cognitively annoying for some users to figure out, but your right that in this case it will be fine, especially since its an internal piece.